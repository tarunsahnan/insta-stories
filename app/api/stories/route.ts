import { NextResponse } from "next/server";
import { stories } from "./stories";
import { NextApiRequest, NextApiResponse } from "next";
import { StoryInterface } from "@/types/StoryInterface";

export async function GET() {
  return NextResponse.json({ stories }, { status: 200 });
}
