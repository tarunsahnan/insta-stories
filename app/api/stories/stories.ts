import { StoryInterface } from "@/types/StoryInterface";

export const stories: StoryInterface[] = [
  {
    userName: "Tarun Sahnan",
    userImage:
      "https://www.wikihow.com/images/thumb/9/90/What_type_of_person_are_you_quiz_pic.png/728px-What_type_of_person_are_you_quiz_pic.png",
    userStory:
      "https://img.freepik.com/free-photo/autumn-leaf-falling-revealing-intricate-leaf-vein-generated-by-ai_188544-9869.jpg?t=st=1715176859~exp=1715180459~hmac=968e24d5f6315f8618e1d0a655f47b7461d4d73cb19ac79ddbd5ddf19fc51813&w=900",
  },
  {
    userName: "Tarun Sahnan",
    userImage:
      "https://www.wikihow.com/images/thumb/9/90/What_type_of_person_are_you_quiz_pic.png/728px-What_type_of_person_are_you_quiz_pic.png",
    userStory:
      "https://img.freepik.com/free-photo/autumn-leaf-falling-revealing-intricate-leaf-vein-generated-by-ai_188544-9869.jpg?t=st=1715176859~exp=1715180459~hmac=968e24d5f6315f8618e1d0a655f47b7461d4d73cb19ac79ddbd5ddf19fc51813&w=900",
  },
  {
    userName: "Tarun Sahnan",
    userImage:
      "https://www.wikihow.com/images/thumb/9/90/What_type_of_person_are_you_quiz_pic.png/728px-What_type_of_person_are_you_quiz_pic.png",
    userStory:
      "https://img.freepik.com/free-photo/autumn-leaf-falling-revealing-intricate-leaf-vein-generated-by-ai_188544-9869.jpg?t=st=1715176859~exp=1715180459~hmac=968e24d5f6315f8618e1d0a655f47b7461d4d73cb19ac79ddbd5ddf19fc51813&w=900",
  },
  {
    userName: "Tarun Sahnan",
    userImage:
      "https://www.wikihow.com/images/thumb/9/90/What_type_of_person_are_you_quiz_pic.png/728px-What_type_of_person_are_you_quiz_pic.png",
    userStory:
      "https://img.freepik.com/free-photo/autumn-leaf-falling-revealing-intricate-leaf-vein-generated-by-ai_188544-9869.jpg?t=st=1715176859~exp=1715180459~hmac=968e24d5f6315f8618e1d0a655f47b7461d4d73cb19ac79ddbd5ddf19fc51813&w=900",
  },
  {
    userName: "Tarun Sahnan",
    userImage:
      "https://www.wikihow.com/images/thumb/9/90/What_type_of_person_are_you_quiz_pic.png/728px-What_type_of_person_are_you_quiz_pic.png",
    userStory:
      "https://img.freepik.com/free-photo/autumn-leaf-falling-revealing-intricate-leaf-vein-generated-by-ai_188544-9869.jpg?t=st=1715176859~exp=1715180459~hmac=968e24d5f6315f8618e1d0a655f47b7461d4d73cb19ac79ddbd5ddf19fc51813&w=900",
  },
  {
    userName: "Tarun Sahnan",
    userImage:
      "https://www.wikihow.com/images/thumb/9/90/What_type_of_person_are_you_quiz_pic.png/728px-What_type_of_person_are_you_quiz_pic.png",
    userStory:
      "https://img.freepik.com/free-photo/autumn-leaf-falling-revealing-intricate-leaf-vein-generated-by-ai_188544-9869.jpg?t=st=1715176859~exp=1715180459~hmac=968e24d5f6315f8618e1d0a655f47b7461d4d73cb19ac79ddbd5ddf19fc51813&w=900",
  },
  {
    userName: "Tarun Sahnan",
    userImage:
      "https://www.wikihow.com/images/thumb/9/90/What_type_of_person_are_you_quiz_pic.png/728px-What_type_of_person_are_you_quiz_pic.png",
    userStory:
      "https://img.freepik.com/free-photo/autumn-leaf-falling-revealing-intricate-leaf-vein-generated-by-ai_188544-9869.jpg?t=st=1715176859~exp=1715180459~hmac=968e24d5f6315f8618e1d0a655f47b7461d4d73cb19ac79ddbd5ddf19fc51813&w=900",
  },
  {
    userName: "Tarun Sahnan",
    userImage:
      "https://www.wikihow.com/images/thumb/9/90/What_type_of_person_are_you_quiz_pic.png/728px-What_type_of_person_are_you_quiz_pic.png",
    userStory:
      "https://img.freepik.com/free-photo/autumn-leaf-falling-revealing-intricate-leaf-vein-generated-by-ai_188544-9869.jpg?t=st=1715176859~exp=1715180459~hmac=968e24d5f6315f8618e1d0a655f47b7461d4d73cb19ac79ddbd5ddf19fc51813&w=900",
  },
];
