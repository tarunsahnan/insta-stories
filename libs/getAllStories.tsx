import { StoryInterface } from "@/types/StoryInterface";

export default async function getAllStories(
  host: string
): Promise<{ stories: StoryInterface[] }> {
  const data = await fetch(`${host}/api/stories`);

  if (!data.ok) {
    throw new Error("Failed to fetch data");
  }

  return data.json();
}
