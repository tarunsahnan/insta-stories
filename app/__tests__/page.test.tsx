import { VitestUtils, expect, test, vi } from "vitest";
import { render, screen } from "@testing-library/react";
import Page from "@/app/page";
import getAllStories from "@/libs/getAllStories";
import { stories } from "../api/stories/stories";

vi.mock("@/libs/getAllStories");
test("Page", () => {
  (getAllStories as any).mockReturnValueOnce(Promise.resolve({ stories }));
  const { container, rerender } = render(<Page />);
  expect(container).toMatchSnapshot();
});
