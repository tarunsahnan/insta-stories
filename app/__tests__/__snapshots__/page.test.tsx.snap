// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Page 1`] = `
<div>
  <div
    class="overflow-hidden"
  >
    <div
      class="flex space-x-2 overflow-x-scroll bg-white border-gray-200 pt-2 px-4 scroll-smooth scrollbar-hide"
    />
  </div>
</div>
`;
